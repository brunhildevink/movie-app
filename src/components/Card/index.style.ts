import styled from 'styled-components'
import { colors, typography } from '../../styles'

const Description = styled.div`
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  line-clamp: 3;
  -webkit-box-orient: vertical;
`

const EpisodeNumber = styled.span`
  width: 32px;
  height: 32px;
  position: absolute;
  top: 0;
  left: 0;
  background: white;
  color: black;
  display: flex;
  align-items: center;
  justify-content: center;

  p {
    font-size: ${typography.fontSizeTextMedium}px;
    font-weight: ${typography.fontWeightBold};
  }
`

const Image = styled.div<{ img?: string }>`
  background: ${({ img }) => (img ? `url(${img}) center center no-repeat` : `${colors.white}`)};
  background-size: cover;
  height: 136px;
  width: 200px;
`

const NotFound = styled.div`
  background: ${colors.white};
  height: 136px;
  width: 200px;
  display: flex;
  align-items: center;
  justify-content: center;
`

const Title = styled.div`
  margin: 20px 0 8px 0;

  p {
    font-size: ${typography.fontSizeTextMedium}px;
    line-height: ${typography.lineHeightTextMedium};
    width: 200px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
`

const Wrapper = styled.button`
  width: 200px;
  position: relative;
  border: 0;
  padding: 0;
  background: transparent;
  text-align: start;
  color: ${colors.white};
  cursor: pointer;
`

export { Description, EpisodeNumber, Image, NotFound, Title, Wrapper }
