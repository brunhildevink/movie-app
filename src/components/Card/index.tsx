import React from 'react'
import styled from 'styled-components'
import Text from '../Text'

interface Props {
  description: string
  episodeNumber: string
  image: string
  title: string
}

const Description = styled.div`
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  line-clamp: 3;
  -webkit-box-orient: vertical;
`

const EpisodeNumber = styled.span`
  width: 32px;
  height: 32px;
  position: absolute;
  top: 0;
  left: 0;
  background: white;
  color: black;
  display: flex;
  align-items: center;
  justify-content: center;
`

const Image = styled.div<{ img: string }>`
  background: ${({ img }) => `url(${img})`} center center no-repeat;
  background-size: cover;
  height: 136px;
  width: 200px;
`

const Title = styled.div`
  p {
    width: 200px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
`

const Wrapper = styled.div`
  width: 200px;
  position: relative;
`

const Card: React.FC<Props> = ({ description, episodeNumber, image, title }) => (
  <Wrapper>
    <Image img={image} />
    <EpisodeNumber>
      <Text.Regular>{episodeNumber}</Text.Regular>
    </EpisodeNumber>
    <Title>
      <Text.Bold>{title}</Text.Bold>
    </Title>
    <Description>
      <Text.Regular>{description}</Text.Regular>
    </Description>
  </Wrapper>
)

export default Card
