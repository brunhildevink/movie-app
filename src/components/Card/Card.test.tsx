import React from 'react'
import renderer from 'react-test-renderer'
import Card from '.'

test('Card renders correctly', () => {
  const tree = renderer.create(<Card id="11234" />).toJSON()
  expect(tree).toMatchSnapshot()
})
