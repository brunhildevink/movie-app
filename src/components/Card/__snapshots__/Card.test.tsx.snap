// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="sc-ftvSup gPyuec"
>
  <div
    className="sc-gKXOVf gqBaPE"
  />
  <span
    className="sc-jSMfEi dbSAYs"
  >
    <p
      className="sc-hKMtZM AqaKO"
    >
      1
    </p>
  </span>
  <div
    className="sc-iBkjds hxeFpw"
  >
    <p
      className="sc-bczRLJ jzQJTL"
    >
      test
    </p>
  </div>
  <div
    className="sc-eCYdqJ hnWALs"
  >
    <p
      className="sc-hKMtZM AqaKO"
    >
      lorem ipsum
    </p>
  </div>
</div>
`;
